lib/twiglet/logger.rb

```diff
@@ -65,16 +70,15 @@ def log(level:, message:)
 
       total_message = {
         service: {
-          name: @service
+          name: @service_name
         },
         "@timestamp": @now.call.iso8601(3),
         log: {
           level: level
         }
       }
-      total_message = total_message.merge(@scoped_properties)
-                                   .merge!(message)
-                                   .then { |log_entry| to_nested(log_entry) }
+      total_message = deep_merge(total_message, to_nested(@scoped_properties))
+      total_message = deep_merge(total_message, to_nested(message))
```

Comment:
I generally like to avoid variable reassignment - I'm pretty sure you can chain these calls!

Reply:
deep_merge is a procedure, so without monkey_patching Hash, no chaining. Possibly use Compose or similar?

Reply:
Any idea how we can do this without reassignment?

Reply:
I'd suggest that the idiomatic Ruby thing here is to make a subclass of Hash that has the deep_merge method (or just monkey patch Hash) and then use something like the below: 

```suggestion
      total_message = total_message.deep_merge(to_nested@default_properties())
                                   .deep_merge(to_nested(message))
```

Reply:
Done!